### PR TITLE
ObjectManager cleanup - Remove usage from AdminNotification module

### PR DESCRIPTION
**Pull request by ihor-sviziev**
### Description (*)
This PR cleans up object manager usage and marks optional arguments as not optional in non API classes.

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/AdminNotification/Block/System/Messages.php [hunk 1]
```diff
@@ -5,45 +5,51 @@
...
     public function __construct(
-        \Magento\Backend\Block\Template\Context $context,
-        \Magento\AdminNotification\Model\ResourceModel\System\Message\Collection\Synchronized $messages,
-        \Magento\Framework\Json\Helper\Data $jsonHelper,
-        array $data = [],
-        \Magento\Framework\Serialize\Serializer\Json $serializer = null
+        TemplateContext $context,
+        Synchronized $messages,
+        JsonDataHelper $jsonHelper,
+        JsonSerializer $serializer,
```
Comment (lbajsarowicz) [inline app/code/Magento/AdminNotification/Block/System/Messages.php:49]: :+1:

File: app/code/Magento/AdminNotification/Block/System/Messages.php [hunk 2]
```diff
@@ -62,15 +68,13 @@ protected function _toHtml()
...
     {
         $items = array_values($this->_messages->getItems());
-        if (isset(
-            $items[0]
-        ) && $items[0]->getSeverity() == \Magento\Framework\Notification\MessageInterface::SEVERITY_CRITICAL
-        ) {
+
+        if (isset($items[0]) && (int)$items[0]->getSeverity() === MessageInterface::SEVERITY_CRITICAL) {
             return $items[0];
         }
```
Comment (lbajsarowicz) [inline app/code/Magento/AdminNotification/Block/System/Messages.php]: I'm just curious if it's not better to use `!empty($items) && current($items)->getSeverity() ...` and then `return current($items)`?

File: app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MarkAsRead.php [hunk 3]
```diff
@@ -23,11 +38,7 @@ public function execute()
...
         if ($notificationId) {
             try {
-                $this->_objectManager->create(
-                    \Magento\AdminNotification\Model\NotificationService::class
-                )->markAsRead(
-                    $notificationId
-                );
+                $this->notificationService->markAsRead($notificationId);
                 $this->messageManager->addSuccessMessage(__('The message has been marked as Read.'));
             } catch (\Magento\Framework\Exception\LocalizedException $e) {
```
Comment (lbajsarowicz) [inline app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MarkAsRead.php]: You can also import this one

**Review by lbajsarowicz — DISMISSED** [review]
@ihor-sviziev Please fix static tests.
There are minor comments, generally speaking :+1:

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6942](https://jira.corp.magento.com/browse/ENGCOM-6942) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by ihor-sviziev** [comment]
@lbajsarowicz could you request the changes in order not to move them to test?

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Leaving at "Request changes" for styling the PHP files.

File: app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MarkAsRead.php (hunk 3 above)
Comment (ihor-sviziev) [reply app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MarkAsRead.php]: Fixed

File: app/code/Magento/AdminNotification/Block/System/Messages.php (hunk 2 above)
Comment (ihor-sviziev) [reply app/code/Magento/AdminNotification/Block/System/Messages.php]: Fixed

**Review by lbajsarowicz — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6942](https://jira.corp.magento.com/browse/ENGCOM-6942) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Review by Nazar65 — CHANGES_REQUESTED** [review]
Hi @engcom-Kilo it would be great if you will squash commits into one.

File: app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MassMarkAsRead.php [hunk 4]
```diff
@@ -24,14 +27,20 @@ class MassMarkAsRead extends Notification
...
     {
         parent::__construct($context);
         $this->inboxModelFactory = $inboxModelFactory;
     }
 
     /**
-     * @return void
+     * @inheritdoc
+     *
+     * @return \Magento\Framework\App\ResponseInterface|\Magento\Framework\Controller\ResultInterface
```
Comment (Nazar65) [inline app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MassMarkAsRead.php]: `@return` is not required if there already `@inheritdoc`

File: app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MassRemove.php [hunk 5]
```diff
@@ -24,14 +27,20 @@ class MassRemove extends Notification
...
     {
         parent::__construct($context);
         $this->inboxModelFactory = $inboxModelFactory;
     }
 
     /**
-     * @return void
+     * @inheritdoc
+     *
+     * @return \Magento\Framework\App\ResponseInterface|\Magento\Framework\Controller\ResultInterface
```
Comment (Nazar65) [inline app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/MassRemove.php]: same here

File: app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/Remove.php [hunk 6]
```diff
@@ -24,23 +27,28 @@ class Remove extends Notification
...
     {
         parent::__construct($context);
         $this->inboxModelFactory = $inboxModelFactory;
     }
 
     /**
-     * @return void
+     * @inheritdoc
+     *
+     * @return \Magento\Framework\App\ResponseInterface|\Magento\Framework\Controller\ResultInterface
```
Comment (Nazar65) [inline app/code/Magento/AdminNotification/Controller/Adminhtml/Notification/Remove.php]: same here

**Review by Nazar65 — CHANGES_REQUESTED** [review]

**Review by Nazar65 — APPROVED** [review]

**Pull request merged**